domains/auth/Auth.API/API/Controllers/TermsController.cs

```diff
@@ -0,0 +1,43 @@
+using API.Models.DTOs;
+using API.Models.Entities;
+using API.Services;
+using API.Utilities;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace API.Controllers;
+
+[Authorize]
+[ApiController]
+public class TermsController : ControllerBase
+{
+    [HttpPut()]
+    [Route("terms/accept")]
+    public async Task<IActionResult> AcceptTermsAsync(
+        IUserDescriptMapper descriptMapper,
+        IUserService userService,
+        [FromBody] AcceptTermsDTO acceptedTermsVersion)
+    {
+        var descriptor = descriptMapper.Map(User) ?? throw new NullReferenceException($"UserDescriptMapper failed: {User}");
+
+        var user = new User();
+
+        if (descriptor.Id is null)
+        {
+            user.Name = descriptor.Name;
+            user.ProviderId = descriptor.ProviderId;
+            user.Tin = descriptor.Tin;
+            user.AllowCPRLookup = descriptor.AllowCPRLookup;
+        }
+        else
+        {
+            user = await userService.GetUserByIdAsync((Guid)descriptor.Id) ?? throw new NullReferenceException($"GetUserByIdAsync() returned null: {descriptor.Id}");
+        }
```

Comment:
```suggestion
        var id = descriptor.Id;
        User user;

        if (id == null)
        {
            user = new User();
            user.Name = descriptor.Name;
            user.ProviderId = descriptor.ProviderId;
            user.Tin = descriptor.Tin;
            user.AllowCPRLookup = descriptor.AllowCPRLookup;
        }
        else
        {
            user = await userService.GetUserByIdAsync(id) ?? throw new NullReferenceException($"GetUserByIdAsync() returned null: {id}");
        }
```

... but `user = new User();` should be using `{}` "constructor".